server/collab/vectors.py

```diff
@@ -0,0 +1,60 @@
+from collections import defaultdict
+
+
+class Vector:
+  @classmethod
+  def match(cls, soure, target):
+    raise NotImplementedError("Method match for vector type {} not "
+                              "implemented".format(cls))
+
+
+class DummyVector(Vector):
+  id = 'dummy'
+  name = 'Dummy'
+
+  @classmethod
+  def match(cls, soure, target):
+    return []
+
+
+class HashVector(Vector):
+  id = 'hash'
+  name = 'Hash'
```

Comment:
The idea is to create a Hash Vector class and make it implement a few common functions all together instead of every hash function. Things like instructions, basic blocks access and others.

Why would you like to remove it ? The HashVec class saves a few copy-n-paste methods I implemented for hash vector classes